.github/workflows/hls.yml

```
@@ -0,0 +1,19 @@
+name: "Haskell Language Server works"
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+ 
+permissions:
+  contents: read
+  
+jobs:
+  test-hls-works:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c
+    - uses: cachix/install-nix-action@daddc62a2e67d1decb56e028c9fa68344b9b7c2a
+      with:
+        nix_path: nixpkgs=channel:nixos-23.11
+    - run: nix develop --accept-flake-config --command bash -c "cabal update; haskell-language-server"
```

```suggestion
      - uses: actions/checkout@v4
      - name: Install Nix with good defaults
        uses: input-output-hk/install-nix-action@v20
        with:
          extra_nix_config: |
            trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ=
            substituters = https://cache.iog.io/ https://cache.nixos.org/
          nix_path: nixpkgs=channel:nixos-unstable
      - uses: cachix/install-nix-action@v18
        with:
          nix_path: nixpkgs=channel:nixos-unstable
      # Make the Nix environment available to next steps
      - uses: rrbutani/use-nix-shell-action@v1
      - run: haskell-language-server
```

Rationale: use same setup as other workflows

---

Solved in: https://github.com/IntersectMBO/cardano-cli/pull/798/commits/05d4c08d687cb54423fec424a66f0db3312501d7
I had to add `cabal update` to the `run` bit too.